tests/unit/test_library_saml.py

```diff
@@ -155,3 +185,44 @@ def test_consumer_charm_emits_event_when_leader():
     assert harness.charm.events[0].metadata_url == relation_data["metadata_url"]
     assert harness.charm.events[0].certificates == tuple(relation_data["x509certs"].split(","))
     assert harness.charm.events[0].endpoints == (slo_ep, sso_ep)
+
+
+def test_provider_charm_doesnt_emit_duplicates():
+    """
+    arrange: set up a charm and update the relation data.
+    act: execute the relation data update again.
+    assert: the relation data is not updated.
+    """
+    sso_ep = saml.SamlEndpoint(
```

Comment:
What do the variable names mean here?

Reply:
sso is short for single sign on. I though it was ok as it's commonly named as such in literature

Reply:
I got the `sso` bit, what about `ep`?

Reply:
Renamed